src/components/Posts/EditPostModal.tsx

```diff
@@ -0,0 +1,190 @@
+import {
+  Dialog,
+  DialogTitle,
+  Button,
+  DialogContent,
+  IconButton,
+  Box,
+  Stack,
+  TextField,
+  Typography,
+} from "@mui/material";
+import CloseIcon from "@mui/icons-material/Close";
+import { useEffect, useState } from "react";
+import { useAppDispatch, useAppSelector } from "../../state/hooks";
+import { EmojiClickData } from "emoji-picker-react";
+import EmojiPickerIconButton from "../Common/EmojiPickerIconButton";
+import UserAvatar from "../Common/UserAvatar";
+import AddPhotoAlternateOutlinedIcon from "@mui/icons-material/AddPhotoAlternateOutlined";
+import formatTimestamp from "../../utilities/formatTimestamp";
+import axios from "axios";
+import { updatePost } from "../../state/slices/postsSlice";
+
+type EditPostModalProps = {
+  onClose: () => void;
+  openModal: boolean;
+  postId: number;
+  originalPostTextContent: string;
+  originalPostTimeStamp: string;
+  prevEditedPostTimestamp: string;
+};
+
+const styles = {
+  avatar: { paddingRight: 1.5 },
+  dialog: {
+    ".MuiDialog-scrollPaper": { alignItems: "flex-start" },
+  },
+  dialogContentContainer: {
+    padding: 2,
+  },
+  dialogTitle: {
+    display: "flex",
+    alignItems: "center",
+  },
+  editPostHeader: {
+    display: "flex",
+  },
+  paperProps: {
+    overflow: "visible",
+    borderRadius: 20,
+  },
+  postActions: {
+    display: "flex",
+    justifyContent: "space-between",
+    marginTop: 1,
+  },
+  textFieldContainer: { width: "100%", paddingTop: 2 },
+  textField: { paddingBottom: 2 },
+  titleText: {
+    fontWeight: "bold",
+    paddingLeft: 3,
+  },
+  userInfoText: {
+    display: "flex",
+    gap: 0.5,
+  },
+};
+
+const EditPostModal = ({
+  onClose,
+  openModal,
+  postId,
+  originalPostTextContent,
+  originalPostTimeStamp,
+  prevEditedPostTimestamp,
+}: EditPostModalProps) => {
+  const [postTextContent, setPostTextContent] = useState("");
+  const user = useAppSelector((state) => state.user);
+  const dispatch = useAppDispatch();
+  const existingPost = useAppSelector((state) =>
+    state.posts.posts.find((post) => post.postId === postId)
+  );
+
+  useEffect(() => {
+    setPostTextContent(originalPostTextContent);
+  }, [originalPostTextContent]);
```

Comment:
We shouldn't need this useEffect, we can set the initial state of `postTextContent` instead. For example, right now it'd be something like:

```
const [postTextContent, setPostTextContent] = useState(originalPostTextContent);
```

But we can take it a step further and just extract that directly from the store object so we can ditch the props too:

```
const existingPost = useAppSelector((state) =>
  state.posts.posts.find((post) => post.postId === postId)
);
const [postTextContent, setPostTextContent] = useState(
  existingPost?.textContent
);
```

Reply:
Bruh, this makes sense, I was wondering how to re-render after I have defined the the `useState` to handle the `originalTextContent`. I will be on the look out for instances I can use the already defined states.

Reply:
Done